### ISSUE TITLE
BeliefPropagation's map_query method use operation="marginalize"

### ISSUE THREAD

**Issue by FatPandao** [pgmpy/pgmpy#1397]
`map_query` in `BeliefPropagation` is calling `self._query` with `operation="marginalize"`, should it be `operation='maximize'`?

```
def map_query(self, variables=None, evidence=None, show_progress=True):
    ...
    final_distribution = self._query(
                variables=variables,
                operation="marginalize",
                evidence=evidence,
                show_progress=show_progress,
            )
    ....
```

**Comment by FatPandao** [comment]
When reading `map_query`, I think the code is computing the joint MAP assignment. For example, X = (A, B, C), then result = argmax_X P(X). But it seems that the code is computing argmax_A P(A),  argmax_B P(B) and argmax_C P(C) which are marginal MAPs.

**Comment by ankurankan** [comment]
@FatPandao Yes, that's correct. It's currently mentioned in the docstring here: https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/inference/ExactInference.py#L349. But we need to fix this behavior to allow the user to specify whether they want a joint assignment or individual.
